src/ImageSharp/Formats/Jpeg/JpegDecoderCore.cs

```diff
@@ -97,7 +98,17 @@ internal sealed class JpegDecoderCore : IRawJpegData, IImageDecoderInternals
         /// <summary>
         /// Scan decoder.
         /// </summary>
-        private HuffmanScanDecoder scanDecoder;
+        private IJpegScanDecoder scanDecoder;
+
+        /// <summary>
+        /// The arithmetic decoding tables.
+        /// </summary>
+        private List<ArithmeticDecodingTable> arithmeticDecodingTables;
+
+        /// <summary>
+        /// The restart interval.
+        /// </summary>
+        private int? resetInterval;
```

Comment:
I don't like this arithmetic coding specific property is leaked in the 'master' class, same goes for reset interval which should only be used inside `scanDecoder` but it's a broad question for some further refactoring - I'll work on it right after scaled decoding PR.